actions/detect-version/src/main.ts

```diff
@@ -80,20 +82,50 @@ async function loadTagVersions(depth: number): Promise<void> {
 
   const knownVersions = await utils.getVersions()
 
-  const latestTag = utils.latestVersionTag(knownVersions)
+  let latestTag = utils.latestVersionTag(knownVersions)
   const latestSha = await git.resolveRef(latestTag)
-  const latestVersion = knownVersions[latestTag]
+  let latestVersion = knownVersions[latestTag]
 
-  const prevTag = await git.previousTag()
+  if (latestVersion && latestVersion.original) {
+    latestVersion.original = utils.sanitizeVersionInput(latestVersion.original)
+    latestVersion = version.parse(latestVersion.original)
```

Comment:
This is quire tricky: you get the latestVersion.original value, then replace it and then replace the _whole value_ of latestVersion with the result of `version.parse()`. The code requires you to read it carefully, to understand that final value of `latestVersion` is defined only by the last line and all previous steps must be completely ignored.

Better to use additional variable here to make it more clear on what is happening:

```typescript
if (latestVersion?.original) {
  const sanitizedStr = utils.sanitizeVersionInput(latestVersion.original)
  latestVersion = version.parse(sanitizedStr)
}
```

Moreover, you still can use constants here. It will look compact if you create small helper function later in the code:
```typescript
const latestVersion = knownVersions[latestTag] ? sanitizeAndParse(knownVersions[latestTag]) : undefined
```

```typescript
function sanitizeAndParse(versionString: string): <don't remember name of the latestVersion type> {
  const sanitized = utils.sanitizeVersionInput(versionString)
  return version.parse(sanitized)
}
```

The same trick is for `curVersion`